src/rpm/builder.rs

```diff
@@ -675,7 +682,7 @@ impl RPMBuilder {
             "4.0-1".to_string(),
         ));
 
-        if matches!(self.compressor, Compressor::Zstd(_)) {
```

Comment:
If we impl `fn as_dependency(&self) -> Option<Dependency>` for the compressor we can simply call `self.requires.extend(compressor.as_dependency())` which is much cleaner.

Reply:
I'm not sure that pattern would generalize well enough to be worthwhile.  In fact this might be the only case where that might be possible.